pinecone/core/openapi/shared/model_utils.py

```diff
@@ -738,12 +740,12 @@ def get_simple_class(input_value):
         return bool
     elif isinstance(input_value, int):
         return int
-    # elif isinstance(input_value, datetime):
-    #    # this must be higher than the date check because
-    #    # isinstance(datetime_instance, date) == True
-    #    return datetime
-    # elif isinstance(input_value, date):
-    #    return date
+    elif isinstance(input_value, datetime):
+        # this must be higher than the date check because
+        # isinstance(datetime_instance, date) == True
+        return datetime
+    elif isinstance(input_value, date):
+        return date
```

Comment:
The date/datetime changes in these generated files reflect some adjustments I had to make in the template code to get correct handling of dates in API responses since the bulk import feature introduces date fields for the first time. Previously we had all of this commented out as a quick-and-dirty fix for an unrelated issue where untyped user metadata strings stored with vectors were sometimes coerced into datetime when fetched if the content looked date-ish.

There's a unit test covering that case to make sure we haven't regressed on that issue while still enabling us to interact with datetime objects when that is our intention.

Reply:
Answers my previous question about the templates, thanks.